### PR TITLE
doc: release_process: no formal point release process

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
There is no formal point release process. Document that fact. If a
formal process ever materializes, we can adjust this.

Fixes: #44087

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: doc/development_process/release_process.rst [hunk 1]
```diff
@@ -284,6 +285,20 @@ governing board.
...
 
+Point releases
+==============
+
+As described :ref:`above <releases>`, the Zephyr Project may create follow-up
+releases from a minor release in the format [Major].[Minor].[Patch Level]. For
+example, following the v3.0 release, the project may create a v3.0.1 release
+with additional patches.
+
+No formal process exists for when these releases are created. They may be
```
Comment (nashif) [inline doc/development_process/release_process.rst:296]: There was a proposal in the security WG to have a release cadence for supported branches (read: dot releases), it was not finalized but the text below is not in line with what discussed. @ceolin  Can we please close on this in the security WG and come up with a proposal, both for regular releases and LTS?

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/development_process/release_process.rst (hunk 1 above)
Comment (ceolin) [reply doc/development_process/release_process.rst:296]: Yes, I'll bring it up again.

Comment (ceolin) [reply doc/development_process/release_process.rst:296]: There is no mention to vulnerabilities here, and vulnerabilities fixes should trigger dot releases. I'll review it with the group.

Comment (mbolivar-nordic) [reply doc/development_process/release_process.rst:296]: Who decides if there are vulnerability fixes? This implies a process from the security WG to the release manager for the branch, right? We need to write that down

**Pull request closed without merging**